meson.build

```diff
@@ -214,7 +224,7 @@ endif
 
 # Windows-only dependencies
 #
-if host_machine.system() == 'windows' or host_machine.system() == 'cygwin'
+if host_machine.system() in ['windows', 'cygwin']
   winsock2_dep = cxx.find_library('ws2_32', required : true)
   winmm_dep = cxx.find_library('winmm', required : true)
 endif
```

Comment:
When I look at this overall sequence, I'm seeing three places where we `if host_machine.system() in ['windows', 'cygwin']` and then set various options.

![2021-02-07_08-18](https://user-images.githubusercontent.com/1557255/107152492-160f4680-691d-11eb-8689-443995f5e21c.png)

Is there any way we can group these entirely toogether or even have them in separate config files?

`include file: '.meson' / 'systems' / host_machine.system()`
`include file: '.meson' / 'cup_families' / host_machine.cpu_family()`

And we wouldn't even need a tree of if X, if Y, if Z, and so on.  Just a future design idea.

Reply:
We would still need to have:

```
if host_machine.system() == <something>
  subdir('path/to/something')
endif
```
So I see no point in doing such thing.

Right now the content of meson scripts is grouped by the topic: conf_data stuff is grouped together and put before other things, external dependencies are grouped together and handled after conf_data, source files and internal dependencies are grouped together and handled afterwards, etc… Splitting this grouping and reorganizing meson scripts to have "topics" in separate directories would be very, very inconvenient.

About `PAGESIZE` - this one should go and be handled internally in `src/cpu/meson.build`, in table/dict configuring per-architecture stuff, but we can't do it during meson migration (we can do it afterwards though, with some code changes and additional testing).

Reply:
>Splitting this grouping and reorganizing meson scripts to have "topics" in separate directories would be very, very inconvenient.

Thanks for explaining the reasoning behind this layout @dreamer.  As I don't have the overall meson picture in my mind (and instead see sequences of diffs "close together" like in the above screenshot), I gets a different view and perception. So that helps.

``` python
if host_machine.system() == <something>
  subdir('path/to/something')
endif

I'm guess it could just be:

`subdir('path/to/something')`

Where `<something>` is a single file that covers all the topics for the `system()`.  And we could use symlinks to connect "windows" and "cygin" together.

But I'm just commenting in general, as I understand that's not the goal of our meson layout and so doesn't make sense for us. 


> About PAGESIZE - this one should go and be handled internally in src/cpu/meson.build, in table/dict configuring per-architecture stuff, but we can't do it during meson migration (we can do it afterwards though, with some code changes and additional testing).

Sounds good - and glad it's on your radar!